pkg/misc/depversion/depversion_test.go

```diff
@@ -0,0 +1,245 @@
+package depversion
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/guacsec/guac/internal/testing/ptrfrom"
+)
+
+func Test_VersionRangeParse(t *testing.T) {
+	testCases := []struct {
+		input  string
+		expect VersionMatchObject
+	}{
+		{
+			// deps.dev test set
+			input: "",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=0"},
+				},
+			},
+		},
+		{
+			input: "1.7.21",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{"=1.7.21"},
+				},
+			},
+		},
+		{
+			input: "3.0.3",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{"=3.0.3"},
+				},
+			},
+		},
+		{
+			input: "<2.0,>=0.12",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{"<2.0,>=0.12"},
+				},
+			},
+		},
+		{
+			input: ">=1.0.0",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=1.0.0"},
+				},
+			},
+		},
+		// not even proper semver... We will conver it to semver
+		{
+			input: ">=1.0.0rc8",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=1.0.0-rc8"},
+				},
+			},
+		},
+		{
+			input: "[1.5.0,)",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=1.5.0"},
+				},
+			},
+		},
+		{
+			input: "[3.0,)",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=3.0"},
+				},
+			},
+		},
+		{
+			input: "^0.11",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=0.11,<1"},
+				},
+			},
+		},
+		{
+			input: "^1",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=1,<2"},
+				},
+			},
+		},
+		{
+			input: "^1.0.25",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=1.0.25,<2"},
+				},
+			},
+		},
+		{
+			input: "^3.0.0 || ^4.0.0",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=3.0.0,<4"},
+					{">=4.0.0,<5"},
```

Comment:
Here too, I think it should be `<3.1` and `<4.1` (increase by 1 the component that comes before the last specified component)